engine/access/rpc/connection/manager.go

```diff
@@ -174,7 +185,13 @@ func (m *Manager) retrieveConnection(grpcAddress string, timeout time.Duration,
 // createConnection creates a new gRPC connection to the remote node at the given address with the specified timeout.
 // If the cachedClient is not nil, it means a new entry in the cache is being created, so it's locked to give priority
 // to the caller working with the new client, allowing it to create the underlying connection.
-func (m *Manager) createConnection(address string, timeout time.Duration, cachedClient *CachedClient, clientType clientType) (*grpc.ClientConn, error) {
```

Comment:
Can we extend godoc here to mention that `networkPubKey` can be nil and what it means?